### PR TITLE
feat: security impr tokenAmountForONE only set in constructor

### PR DESCRIPTION
**Pull request by manuelpolzhofer**
- no file method for variables which should only be set once
- any recommendations for a better variable name `tokenAmountForONE`

### CODE REVIEW

File: src/core/lender/assessor/base.sol [hunk 1]
```diff
@@ -89,6 +91,7 @@ contract BaseAssessor is Math, Auth {
     }
 
     function calcTokenPrice(address tranche) public returns (uint) {
+        require(tranche  == junior || tranche == senior);
```
Comment (manuelpolzhofer) [inline src/core/lender/assessor/base.sol]: I think the assessor should only give you numbers for senior and junior.

Theoretically, somebody could implement the interface in a contract and we would return a number

Comment (lucasvo) [reply src/core/lender/assessor/base.sol]: yes

File: src/core/lender/assessor/default.sol [hunk 2]
```diff
@@ -20,6 +20,8 @@ import "tinlake-math/interest.sol";
 
 contract DefaultAssessor is BaseAssessor, Interest {
 
+    constructor(uint tokenAmountForONE) BaseAssessor(tokenAmountForONE) public {}
+
     // accrueTrancheInterest can implement different interest models
     function accrueTrancheInterest(address tranche_) public returns (uint) {
```
Comment (charleenfei) [inline src/core/lender/assessor/default.sol]: should we also do the same address check here?

Comment (manuelpolzhofer) [reply src/core/lender/assessor/default.sol]: I will add it

**Pull request merged**